mteb/models/openai_models.py

```diff
@@ -15,27 +15,48 @@
 
 
 class OpenAIWrapper(Wrapper):
-    def __init__(self, model_name: str, embed_dim: int | None = None, **kwargs) -> None:
+    def __init__(
+        self,
+        model_name: str,
+        max_tokens: int,
+        tokenizer_name: str = "cl100k_base",  # since all models use this tokenizer now
+        embed_dim: int | None = None,
+        **kwargs,
+    ) -> None:
         requires_package(self, "openai", "Openai text embedding")
         from openai import OpenAI
 
         self._client = OpenAI()
         self._model_name = model_name
         self._embed_dim = embed_dim
+        self._max_tokens = max_tokens
+        self._tokenizer_name = tokenizer_name
 
     def encode(self, sentences: list[str], **kwargs: Any) -> np.ndarray:
         requires_package(self, "openai", "Openai text embedding")
+        requires_package(self, "tiktoken", "Tiktoken package")
```

Comment:
We should add this to the pyproject.toml to ensure that version are kept compatible

Reply:
We could probably bump them together such that have them together in al optional dependency set: `pip install mteb[openai]`

@Samoed what do you think?

Reply:
(we should probably also change requires_package) to have "tiktoken", "tiktoken package", "{dependency group}").

Reply:
I agree

Reply:
I do think this is a good idea, but don't really know how to handle this.
What should I do ?

Reply:
You should update dependencies in `pyproject.toml`

Reply:
Would this be okay?
https://github.com/embeddings-benchmark/mteb/pull/1526/commits/23600ea2e6a717c4bee72f86ef6f6ef2e460d5e4

Reply:
perfect